core-bundle/src/Resources/contao/library/Contao/System.php

```diff
@@ -521,6 +521,19 @@ public static function loadLanguageFile($strName, $strLanguage=null, $blnNoCache
 			}
 		}
 
+		// Set MSC.textDirection (see #3360)
+		if ('default' === $strName)
+		{
+			if (empty($GLOBALS['TL_LANG']['MSC']['textDirection']))
+			{
+				$GLOBALS['TL_LANG']['MSC']['textDirection'] = (\ResourceBundle::create($strLanguage, 'ICUDATA', true)['layout']['characters'] ?? null) === 'right-to-left' ? 'rtl' : 'ltr';
+			}
+			else
+			{
+				@trigger_error('Using "MSC.textDirection" to set the text direction will no longer work in Contao 5.0. To override the text direction create a custom "fe_page" template instead.', E_USER_DEPRECATED);
+			}
+		}
+
```

Comment:
I think this would trigger the deprecation now if two different languages get loaded one after the other as the first one sets `$GLOBALS['TL_LANG']['MSC']['textDirection']`.

We should only trigger the deprecation if `empty($GLOBALS['TL_LANG']['MSC']['textDirection'])` is true in the beginning of the `loadLanuageFile` method I think.

Reply:
It needs to be triggered after the language files are actually loaded, otherwise it will always be empty. 

When another language gets loaded, the whole `$GLOBALS['TL_LANG']` array is replaced anyway, so this should be fine. Also I have updated the code so that the deprecation is only triggered on first load of the language, i.e. when it is not cached yet. Otherwise `$GLOBALS['TL_LANG']['MSC']['textDirection']` will not empty, since it's already loaded for the current language.

Reply:
So for example:

```php
System::loadLanguageFile('default', 'fa'); // $GLOBALS['TL_LANG']['MSC']['textDirection'] is now available
System::loadLanguageFile('default', 'fa', true); // no deprecation triggered now
```

Reply:
Oh wait, now the deprecation is never triggered 🙈

Reply:
> When another language gets loaded, the whole `$GLOBALS['TL_LANG']` array is replaced anyway

I don’t think it does. `$GLOBALS['TL_LANG']` never gets cleared and the language files write to the label keys directly.

Reply:
Yes you are right. But then the only way to test whether `MSC.textDirection` is set by a custom translation is to always unset `MSC.textDirection` first before loading the language. Which is fine since we will always set it afterwards anyway. I will adjust the PR accordingly.

Reply:
Adjusted in 25d394004